### PR TITLE
Add dnn model with feature columns using the census data in the model zoo.

### PR DESCRIPTION
**Pull request by brightcoder01**
- Add this model and recordio_gen to verify the training process using ElasticDL embedding column. The verification result is passed and we can get the expected test accuracy value.
- Write this sample model with using Sequential/Functional/Subclass way. In the next step, we will summarize the general pattern/standard of the model definition.

### CODE REVIEW

**Comment by skydoorkai** [comment]
Any unit test to put into `python/tests/example_test.py`?

**Comment by brightcoder01** [comment]
> python

Good suggestion.
The downloading of the source data from https://archive.ics.uci.edu/ml/machine-learning-databases/adult/adult.data is not stable. If we add the census_recordio_gen.py into the Dockerfile.dev, the image build may hang for some time when the network is unstable. So I choose not to put the verification using census dataset into the unit test.
I'll add a test case using the small heart dataset instead to cover the ElasticDL embedding column.

**Comment by skydoorkai** [comment]
> > python
> 
> Good suggestion.
> The downloading of the source data from https://archive.ics.uci.edu/ml/machine-learning-databases/adult/adult.data is not stable. If we add the census_recordio_gen.py into the Dockerfile.dev, the image build may hang for some time when the network is unstable. So I choose not to put the verification using census dataset into the unit test.
> I'll add a test case using the small heart dataset instead to cover the ElasticDL embedding column.

We can generate a small dataset with random data in  `create_recordio_file` (test_utils.py), similar to other tests.

File: model_zoo/census_dnn_model/census_feature_columns.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import tensorflow as tf
+
+from elasticdl.python.elasticdl.feature_column import feature_column
```
Comment (workingloong) [inline model_zoo/census_dnn_model/census_feature_columns.py:3]: The feature_column.py only contains `embedding_column`. Maybe it better to rename feature_column.py to embedding_column.py?

File: elasticdl/python/data/recordio_gen/census_recordio_gen.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+    feature = tf.train.Feature(
+        int64_list=tf.train.Int64List(value=[data_series[LABEL_KEY]])
+    )
+    features[LABEL_KEY] = feature
+
+    return features
+
+
+def convert_to_recordio_files(data_frame, dir_name, records_per_shard):
```
Comment (terrytangyuan) [inline elasticdl/python/data/recordio_gen/census_recordio_gen.py:94]: Maybe reuse `data.odps_recordio_conversion_utils.write_recordio_shards_from_iterator()`?

**Review by skydoorkai — APPROVED** [review]
